lib/dockly/deb.rb

```diff
@@ -214,28 +220,41 @@ def post_startup_script
     scripts = ["#!/bin/bash"]
     bb = Dockly::BashBuilder.new
     scripts << bb.normalize_for_dockly
-    if get_registry
-      scripts << bb.registry_import(docker.repo, docker.tag)
-    elsif docker
-      if docker.s3_bucket.nil?
-        docker_output = File.join(docker.package_dir, docker.export_filename)
-        if docker.tar_diff
-          scripts << bb.file_diff_docker_import(docker.import, docker_output, docker.name, docker.tag)
-        else
-          scripts << bb.file_docker_import(docker_output, docker.name, docker.tag)
+
+    if docker
+      if (registry = docker.registry)
+        if registry.is_a?(Dockly::Docker::ECR)
+          scripts << bb.auth_ecr(registry.server_address)
         end
+        scripts << bb.registry_import(docker.repo, docker.tag)
       else
-        if docker.tar_diff
-          scripts << bb.s3_diff_docker_import(docker.import, docker.s3_url, docker.name, docker.tag)
-        else
-          scripts << bb.s3_docker_import(docker.s3_url, docker.name, docker.tag)
-        end
+        scripts += collect_non_registry_scripts(bb)
```

Comment:
does += work on arrays of strings?
```[3] pry(main)> [] += 1
SyntaxError: unexpected tOP_ASGN, expecting end-of-input
[] += 1
   ^~```

Reply:
This should work as long as `collect_non_registry_scripts` returns an array too:
```
irb(main):001:0> ary = []
=> []
irb(main):002:0> ary << 'string'
=> ["string"]
irb(main):003:0> ary += ['another string', 'more strings']
=> ["string", "another string", "more strings"]
```
https://github.com/swipely/dockly/pull/97/files#diff-062c481ae9dec7be0b9b8633a424f21dade0c8bbbd750a57636c4d835e68e710R239

Reply:
```
[3] pry(main)> [] += [1]
SyntaxError: unexpected tOP_ASGN, expecting end-of-input
[] += [1]
   ^~
[3] pry(main)> a = []
=> []
[4] pry(main)> a += [1]
=> [1]
```
aha, += works on variables, not objects. Ruby is weird! Good stuff.